documentation/modules/cruise-control/proc-generating-optimization-proposals.adoc

```diff
@@ -49,6 +49,8 @@ spec:
     - RackAwareGoal
     - ReplicaCapacityGoal
 ----
++
+In the above example, if `RackAwareGoal` and `ReplicaCapacityGoal` are configured as the only master optimization goals (in `Kafka.spec.cruiseControl.config.goals`), you must add `skip_hard_goal_check=true` to the Cruise Control deployment configuration. Otherwise an error will occur. 
```

Comment:
So I don't really understand this point. The issue would come if the configured `hard.goals` had more than those two goals in? If master goals had these two in and so did `hard.goals` then you would be fine. 

This highlights another issue that we should highlight that if a user changes master goals they should also make sure that they make sure `hard.goals` is still a subset of the new master goals or strange errors will occur (in the CC release we are using, the latest release adds more checks that would fail to start if the user messed this up).

Reply:
This was issue that I mentioned, 'cause I was curious about it -> if you set for example these two `hard.goals` and then apply the KafkaRebalance, again with these two `hard.goals` the error will occur -> it shows something like `tried to find [all hard goals] in [rackAwareGoal, replicaCapacityGoal]` or something like that.

Reply:
And it's because of me that @laidan6000 add this point, because I found it when I was testing the documentation.

Reply:
The issue is that the default `hard.goals` list has 5 (I think) goals in it. So if you set master goals to a list that does not include all those `hard.goals` then you will fail every rebalance (unless you set skip hard goals) and every periodic cached proposal calculation will fail as well.

This will be fixed in the next strimzi release becasue I filed an issue for it and it has been fixed in the latest release and we can add some additional sanity checks as well if we need to. 

**For the current release** we need to document that the user **must** make sure that if they change the master goals that they also make sure that `hard.goals` is a subset of the master goals they set.

Reply:
Okay, thanks for the explanation and the issue you created for CC team. I tested it like (because I was) a customer and it wasn't clear for me for the first look.

So from my POV it would be nice to have some note about what you mentioned.

Reply:
I removed the sentence referred to above in line 53.

I added the following note to the master optimization goals section:

**NOTE:** If you change the inherited master optimization goals, you must ensure that the hard goals, if configured in the `hard.goals` property in `Kafka.spec.cruiseControl.config`, are a subset of the master optimization goals that you configured. Otherwise, errors will occur when generating optimization proposals.

Reply:
Good that you have added the note, but there is not enough context in this sentence for it to be meaningful. For the sentence to be true, a user would have had to break the rule we just outlined in the note in the master goals section? I think it would be better to remove this sentence all together.

Reply:
OK. I have actually removed the sentence. Glad you're happy with the new NOTE.